[BUG] Unstable test for GES

Sometimes one of tests for GES fails. Here's the log:
```python
FAILED pgmpy/tests/test_estimators/test_GES.py::test_child_model - AssertionError: assert 3 <= 2
 +  where 3 = len(({('BirthAsphyxia', 'Disease'), ('CO2', 'CO2Report'), ('CardiacMixing', 'HypDistrib'), ('CardiacMixing', 'HypoxiaInO2'), ('ChestXray', 'XrayReport'), ('Disease', 'Age'), ...} - {('Age', 'Disease'), ('Age', 'Sick'), ('BirthAsphyxia', 'Disease'), ('CO2', 'CO2Report'), ('CO2', 'LungParench'), ('CO2Report', 'CO2'), ...}))
 +    where {('BirthAsphyxia', 'Disease'), ('CO2', 'CO2Report'), ('CardiacMixing', 'HypDistrib'), ('CardiacMixing', 'HypoxiaInO2'), ('ChestXray', 'XrayReport'), ('Disease', 'Age'), ...} = set(OutEdgeView([('BirthAsphyxia', 'Disease'), ('HypDistrib', 'LowerBodyO2'), ('HypoxiaInO2', 'LowerBodyO2'), ('HypoxiaInO2', 'RUQO2'), ('CO2', 'CO2Report'), ('ChestXray', 'XrayReport'), ('Grunting', 'GruntingReport'), ('Disease', 'Age'), ('Disease', 'LVH'), ('Disease', 'DuctFlow'), ('Disease', 'CardiacMixing'), ('Disease', 'LungParench'), ('Disease', 'LungFlow'), ('Disease', 'Sick'), ('LVH', 'LVHreport'), ('DuctFlow', 'HypDistrib'), ('CardiacMixing', 'HypDistrib'), ('CardiacMixing', 'HypoxiaInO2'), ('LungParench', 'HypoxiaInO2'), ('LungParench', 'CO2'), ('LungParench', 'ChestXray'), ('LungParench', 'Grunting'), ('LungFlow', 'ChestXray'), ('Sick', 'Grunting'), ('Sick', 'Age')]))
 +      where OutEdgeView([('BirthAsphyxia', 'Disease'), ('HypDistrib', 'LowerBodyO2'), ('HypoxiaInO2', 'LowerBodyO2'), ('HypoxiaInO2', 'RUQO2'), ('CO2', 'CO2Report'), ('ChestXray', 'XrayReport'), ('Grunting', 'GruntingReport'), ('Disease', 'Age'), ('Disease', 'LVH'), ('Disease', 'DuctFlow'), ('Disease', 'CardiacMixing'), ('Disease', 'LungParench'), ('Disease', 'LungFlow'), ('Disease', 'Sick'), ('LVH', 'LVHreport'), ('DuctFlow', 'HypDistrib'), ('CardiacMixing', 'HypDistrib'), ('CardiacMixing', 'HypoxiaInO2'), ('LungParench', 'HypoxiaInO2'), ('LungParench', 'CO2'), ('LungParench', 'ChestXray'), ('LungParench', 'Grunting'), ('LungFlow', 'ChestXray'), ('Sick', 'Grunting'), ('Sick', 'Age')]) = <pgmpy.models.DiscreteBayesianNetwork.DiscreteBayesianNetwork object at 0x7faef76d07f0>.edges
 +    and   {('Age', 'Disease'), ('Age', 'Sick'), ('BirthAsphyxia', 'Disease'), ('CO2', 'CO2Report'), ('CO2', 'LungParench'), ('CO2Report', 'CO2'), ...} = set(OutEdgeView([('LowerBodyO2', 'HypoxiaInO2'), ('HypoxiaInO2', 'LowerBodyO2'), ('HypoxiaInO2', 'RUQO2'), ('HypoxiaInO2', 'CardiacMixing'), ('ChestXray', 'XrayReport'), ('HypDistrib', 'DuctFlow'), ('DuctFlow', 'HypDistrib'), ('DuctFlow', 'Disease'), ('LungParench', 'CO2'), ('LungParench', 'Disease'), ('LungParench', 'ChestXray'), ('LungParench', 'Grunting'), ('CO2', 'CO2Report'), ('CO2', 'LungParench'), ('LungFlow', 'Disease'), ('LungFlow', 'ChestXray'), ('Disease', 'DuctFlow'), ('Disease', 'LVH'), ('Disease', 'Sick'), ('Disease', 'BirthAsphyxia'), ('Disease', 'LungParench'), ('Disease', 'Age'), ('Disease', 'CardiacMixing'), ('Disease', 'LungFlow'), ('Grunting', 'GruntingReport'), ('LVH', 'LVHreport'), ('LVH', 'Disease'), ('LVHreport', 'LVH'), ('CardiacMixing', 'HypoxiaInO2'), ('CardiacMixing', 'Disease'), ('Age', 'Disease'), ('Age', 'Sick'), ('Sick', 'Disease'), ('Sick', 'Age'), ('Sick', 'Grunting'), ('CO2Report', 'CO2'), ('BirthAsphyxia', 'Disease'), ('RUQO2', 'HypoxiaInO2')]))
 +      where OutEdgeView([('LowerBodyO2', 'HypoxiaInO2'), ('HypoxiaInO2', 'LowerBodyO2'), ('HypoxiaInO2', 'RUQO2'), ('HypoxiaInO2', 'CardiacMixing'), ('ChestXray', 'XrayReport'), ('HypDistrib', 'DuctFlow'), ('DuctFlow', 'HypDistrib'), ('DuctFlow', 'Disease'), ('LungParench', 'CO2'), ('LungParench', 'Disease'), ('LungParench', 'ChestXray'), ('LungParench', 'Grunting'), ('CO2', 'CO2Report'), ('CO2', 'LungParench'), ('LungFlow', 'Disease'), ('LungFlow', 'ChestXray'), ('Disease', 'DuctFlow'), ('Disease', 'LVH'), ('Disease', 'Sick'), ('Disease', 'BirthAsphyxia'), ('Disease', 'LungParench'), ('Disease', 'Age'), ('Disease', 'CardiacMixing'), ('Disease', 'LungFlow'), ('Grunting', 'GruntingReport'), ('LVH', 'LVHreport'), ('LVH', 'Disease'), ('LVHreport', 'LVH'), ('CardiacMixing', 'HypoxiaInO2'), ('CardiacMixing', 'Disease'), ('Age', 'Disease'), ('Age', 'Sick'), ('Sick', 'Disease'), ('Sick', 'Age'), ('Sick', 'Grunting'), ('CO2Report', 'CO2'), ('BirthAsphyxia', 'Disease'), ('RUQO2', 'HypoxiaInO2')]) = <pgmpy.base.PDAG.PDAG object at 0x7faef4117670>.edges
= 1 failed, 807 passed, 13 skipped, 1408 deselected, 8 xfailed, 3587 warnings in 686.60s (0:11:26) =
Error: Process completed with exit code 1.
```

So I ran that same test snippet 

with different python hashseeds on my laptop
and it gives different edges for some values

```

=== PYTHONHASHSEED=7 ===
pgmpy/tests/test_estimators/test_GES.py::test_child_model FAILED         [100%]
FAILED pgmpy/tests/test_estimators/test_GES.py::test_child_model - AssertionE...
================== 1 failed, 2 warnings in 89.19s (0:01:29) ===================
```
FAILED at PYTHONHASHSEED=7, works for 0-6, may be different in you rsystem
It's probably because
`current_model.nodes()` returns nodes in hash-dependent order. When multiple operations in the forward/backward phases have the same score delta, np.argmax picks the first, so different node orderings produce different tie-breaking, graphs.
Not necessarily a problem I think, the test just seems to have low tolerance

--- a/pgmpy/causal_discovery/GES.py
+++ b/pgmpy/causal_discovery/GES.py
@@ -131,7 +131,7 @@ class GES(_ScoreMixin, _BaseCausalDiscovery):
         """
         Return all edges that can be considered for deletion.
         """
-        return list(current_model.edges())
+        return sorted(current_model.edges())
 
     def insert(
         self,
@@ -256,7 +256,7 @@ class GES(_ScoreMixin, _BaseCausalDiscovery):
         # Step 2: Forward phase. Iteratively add edges till score stops improving.
         while True:
             potential_edges = []
-            for u, v in combinations(current_model.nodes(), 2):
+            for u, v in combinations(sorted(current_model.nodes()), 2):
                 if not current_model.has_edge(u, v) and not current_model.has_edge(v, u):
                     potential_edges.append((u, v))
                     potential_edges.append((v, u))
@@ -371,7 +371,7 @@ class GES(_ScoreMixin, _BaseCausalDiscovery):
         # Step 4: Turning phase. Iteratively reorient edges till score stops improving.
         while True:
             potential_turns = []
-            for u, v in current_model.edges():
+            for u, v in sorted(current_model.edges()):
                 potential_turns.append((v, u))
 
             score_deltas = np.zeros(len(potential_turns))

--- a/pgmpy/estimators/GES.py
+++ b/pgmpy/estimators/GES.py
@@ -84,7 +84,7 @@ class GES(StructureEstimator):
         """
         legal_edges: list[tuple[Hashable, Hashable]] = []
 
-        for u, v in combinations(current_model.nodes(), 2):
+        for u, v in combinations(sorted(current_model.nodes()), 2):
             # Nodes must not be adjacent in any direction
             if not current_model.has_edge(u, v) and not current_model.has_edge(v, u):
                 legal_edges.append((u, v))
@@ -99,7 +99,7 @@ class GES(StructureEstimator):
         """
         Return all edges that can be considered for deletion.
         """
-        return list(current_model.edges())
+        return sorted(current_model.edges())
 
     def _legal_edge_turns(
         self,
@@ -110,7 +110,7 @@ class GES(StructureEstimator):
         """
         legal_turns: list[tuple[Hashable, Hashable]] = []
 
-        for u, v in current_model.edges():
+        for u, v in sorted(current_model.edges()):
             legal_turns.append((v, u))
 
         return legal_turns

